docs/src/pages/index.tsx

```diff
@@ -5,9 +5,53 @@
 
 import Layout from "@theme/Layout";
 import React from "react";
+import { v4 as uuidv4 } from "uuid";
 
+import appInsights from "@site/src/appInsights";
 import { Homepage } from "@site/src/components/home";
 
+/**
+ * Helper function to get a cookie value
+ * @param name - The name of the cookie
+ * @returns - The value of the cookie
+ */
+function getCookie(nam: string): string | undefined {
+	const match = document.cookie.match(new RegExp(`(^| )${name}=([^;]+)`));
+	return match ? decodeURIComponent(match[2]) : undefined;
+}
+
+/**
+ * Helper function to set a cookie value
+ * @param name - The name of the cookie
+ * @param value - The value to set
+ * @param days - The number of days until the cookie expires
+ */
+function setCookie(name: string, value: string, days: number): void {
+	if (!/^[\w-]+$/.test(name)) {
+		throw new Error(
+			`Invalid cookie name: "${name}". Only alphanumeric characters, hyphens, and underscores are allowed.`,
+		);
+	}
+	const expires = new Date(Date.now() + days * 86400000).toUTCString();
+	// eslint-disable-next-line unicorn/no-document-cookie
+	document.cookie = `${name}=${encodeURIComponent(value)}; expires=${expires}; path=/; SameSite=Strict`;
+}
+
+/**
+ * Initialize Application Insights and set the user ID.
+ */
```

Comment:
Nit (since this is inline code)
```suggestion
// Initialize Application Insights and set the user ID.
```